tests/testsuite/check.rs

```diff
@@ -772,3 +778,116 @@ fn error_from_deep_recursion() -> Result<(), fmt::Error> {
 
     Ok(())
 }
+
+#[cargo_test]
+fn workspace_rustc_wrapper_affects_all_workspace_members() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [workspace]
+            members = ["bar", "baz"]
+        "#,
+        )
+        .file("bar/Cargo.toml", &basic_manifest("bar", "0.1.0"))
+        .file("bar/src/lib.rs", "pub fn bar() {}")
+        .file("baz/Cargo.toml", &basic_manifest("baz", "0.1.0"))
+        .file("baz/src/lib.rs", "pub fn baz() {}")
+        .build();
+
+    p.cargo("check")
+        .env("WORKSPACE_RUSTC_WRAPPER", "rustc-echo-wrapper")
+        .with_stdout_contains("WRAPPER CALLED: rustc --crate-name bar [..]")
+        .with_stdout_contains("WRAPPER CALLED: rustc --crate-name baz [..]")
+        .run();
+}
+
+#[cargo_test]
+fn workspace_rustc_wrapper_includes_path_deps() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [project]
+            name = "foo"
+            version = "0.1.0"
+            authors = []
+
+            [workspace]
+            members = ["bar"]
+
+            [dependencies]
+            baz = { path = "baz" }
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .file("bar/Cargo.toml", &basic_manifest("bar", "0.1.0"))
+        .file("bar/src/lib.rs", "pub fn bar() {}")
+        .file("baz/Cargo.toml", &basic_manifest("baz", "0.1.0"))
+        .file("baz/src/lib.rs", "pub fn baz() {}")
+        .build();
+
+    p.cargo("check --workspace")
+        .env("WORKSPACE_RUSTC_WRAPPER", "rustc-echo-wrapper")
+        .with_stdout_contains("WRAPPER CALLED: rustc --crate-name foo [..]")
+        .with_stdout_contains("WRAPPER CALLED: rustc --crate-name bar [..]")
+        .with_stdout_contains("WRAPPER CALLED: rustc --crate-name baz [..]")
+        .run();
+}
+
+#[cargo_test]
+fn workspace_rustc_wrapper_respects_primary_units() {
```

Comment:
Same as above, do we expect the workspace wrapper to not show up in output when we specify `-p UNIT`?

Reply:
In this particular example, it seems correct, because `baz` is not built at all.  If `baz` is a dependency of `bar`, then I would expect it to run for `baz` too.